lib/resources/oracledb_session.rb

```diff
@@ -56,11 +56,11 @@ def query(q)
         p = :parse_csv_result
       else
         bin = @sqlplus_bin
-        opts = "SET MARKUP HTML ON\nSET FEEDBACK OFF"
+        opts = "SET MARKUP HTML ON;\nSET FEEDBACK OFF;"
         p = :parse_html_result
       end
 
-      command = "echo \"#{opts}\n#{verify_query(escaped_query)}\nEXIT\" | #{bin} -s #{@user}/#{@password}@//#{@host}:#{@port}/#{@service}"
+      command = "echo \"#{opts}\n#{verify_query(escaped_query)};\nEXIT\" | #{bin} -s \"#{@user}\"/\"#{@password}\"@//#{@host}:#{@port}/\"#{@service}\""
```

Comment:
Have we tested this with a query where the user supplies a semicolon-terminated query? What happens if the query therefore ends up with `;;`?

I'd rather see us only add the semicolon if the user doesn't supply it. Something like this:

```ruby
query = verify_query(escaped_query)
query += ';' unless query.end_with?(';')
command = %{echo "#{opts}\n#{query}\nEXIT" | #{bin} -s "#{@user}"/"#{@password}"@//#{@host}:#{@port}/"#{@service}"}
```

Note the use of `%{ }` to avoid having to do all that escaping of the double-quotes... should make it much more readable as well.